crates/graphql/src/main.rs

```diff
@@ -35,38 +35,47 @@ fn graphiql(uri: String) -> impl Fn() -> HttpResponse + Clone {
     }
 }
 
-async fn graphql(
-    st: web::Data<Arc<Schema>>,
-    data: web::Json<GraphQLRequest>,
-) -> Result<HttpResponse, Error> {
-    let json = web::block(move || {
-        let res = data.execute_sync(&st, &());
-        serde_json::to_string(&res)
-    })
-    .await?;
-    Ok(HttpResponse::Ok()
-        .content_type("application/json")
-        .body(json))
+fn graphql(
+    db_pool: &Arc<Pool>,
+) -> impl Fn(
+    web::Data<Arc<Schema>>,
+    web::Json<GraphQLRequest>,
+) -> Pin<Box<dyn Future<Output = Result<HttpResponse, Error>> + Send>>
++ Clone {
+    let db_pool = Arc::clone(db_pool);
+
+    move |st, data| {
+        let db_connection = db_pool.get().unwrap();
```

Comment:
@ray-kast do we not need to close the connection after getting the res? or is that handled with the function exiting?